BrainPortal/app/controllers/users_controller.rb

```diff
@@ -415,35 +415,51 @@ def send_password #:nodoc:
   end
 
   def push_keys #:nodoc:
-    @user = User.where(:id => params[:id]).first
-    cb_error "You don't have permission to update this user.", :redirect => start_page_path unless edit_permission?(@user)
+    @user = User.find(params[:id])
+    cb_error "You don't have permission to update this user.", :redirect => user_path(@user) unless edit_permission?(@user)
+
+    push_bids        = params[:push_keys_to].presence
+    bourreau_to_push = Bourreau.find_all_accessible_by_user(@user).where(:id => push_bids).to_a
+    ssh_key          = @user.ssh_key rescue nil
 
-    push_bids = params[:push_keys_to].presence
-    cb_notice "No servers selected.", :redirect => user_path(@user) if push_bids.blank?
-    ssh_key   = @user.ssh_key rescue nil
-    cb_notice "No user SSH key exists yet.", :redirect => :show if ! ssh_key
+    cb_error "No servers selected (or accessible by user).", :redirect => user_path(@user) if bourreau_to_push.empty?
+    cb_error "No user SSH key exists yet.", :redirect => :show     if ! ssh_key
```

Comment:
Why is the indentation messed up? And why do we have both `redirect => user_path(@user)` and `redirect => :show` ?